wagtail/core/migrations/0063_collection_management_permissions.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+from django.db import migrations
+
+
+def grant_instance_level_collection_management_permissions(apps, schema_editor):
+    Collection = apps.get_model('wagtailcore.Collection')
+    GroupCollectionPermission = apps.get_model('wagtailcore.GroupCollectionPermission')
+    Permission = apps.get_model('auth.Permission')
+
+    # Give all groups who currently manage collections permission to manage root collection
+    root_collection = Collection.objects.filter(depth=1).order_by('path').first()
+    stock_permissions = Permission.objects.filter(
+        content_type__app_label='wagtailcore',
+        content_type__model='collection',
+        codename__in=['add_collection', 'change_collection', 'delete_collection']
+    ).prefetch_related('group_set').all()
+
+    for perm in stock_permissions:
+        for group in perm.group_set.all():
+            gcp = GroupCollectionPermission(
+                group_id=group.id,
+                collection_id=root_collection.id,
+                permission_id=perm.id
+            )
+            gcp.full_clean()
+            gcp.save()
```

Comment:
@cnk We should avoid cleaning here, as there's no handling for `ValidationError` in migrations. We just need to be confident that these changes do what we expect. We could maybe shorten to...
```suggestion
            GroupCollectionPermission.objects.create(
                group_id=group.id,
                collection_id=root_collection.id,
                permission_id=perm.id
            )
```

Reply:
That is much cleaner. I have incorporated the direct `create` in my other changes to the migration.